File: Sexta_Generacion_CPCFI/ICPC_31-08-24/A.py
n = int(input())
k = int(input())

aux = k//n

while True:
    if k == n:
        print(1)
        break
    if n == 1:
        print(k)
        break
    if k-(n*aux) < n:
        aux -= 1
    else:
        print(aux)
        break

File: Sexta_Generacion_CPCFI/ICPC_31-08-24/A.cpp
#include <iostream>
using namespace std;

void solve() {
    int n, k;
    cin >> n;
    cin >> k;
    
    int aux = k / n;
    
    if(n == k) {
        cout << 1 << '\n';
        return;
    }
    if(n == 1) {
        cout << k << '\n';
        return;
    }
    while(1) {
        if((k - (n*aux)) < n) {
            aux--;
        }
        else {
            cout << aux << '\n';
            break;
        }
    }
}

int main() {
    solve();
}
